backends/lean/Aeneas/Progress/ProgressStar.lean

```diff
@@ -0,0 +1,326 @@
+      matcher := ma.matcherName,
+      motive := ma.motive,
+      uLevels := ma.matcherLevels.toList,
+      params := ma.params
+    }
+  else 
+    /- e.withApp fun f args => logWarning s!"{f} {args}" -/
+    return none
+
+def Info.toExpr(info: Info): Expr :=
+  let fn := Expr.const info.matcher info.uLevels
+  let args := info.params ++
+    #[info.motive] ++
+    info.discrs.map (·.toExpr) ++
+    info.branches.map (·.toExpr)
+  mkAppN fn args
+
+end Bifurcation/- }}} -/
+
+private partial 
+def traverseProgram {α} [Monad m] [MonadError m] [Nonempty (m α)] 
+  /- [MonadLog m] [AddMessageContext m] [MonadOptions m] -/ 
+  [MonadLiftT MetaM m] [MonadControlT MetaM m] 
+  (onResult: Expr -> m α)
+  (onBind: Expr -> Name -> m α -> m α)
+  (onBif: Bifurcation.Info -> Array (Array Name × m α) -> m α)
+  (e: Expr)
+: m α := do
+  let e <- (Utils.normalizeLetBindings e)
+  if let .const ``Bind.bind .. := e.getAppFn then
+    let #[_m, _self, _α, _β, value, cont] := e.getAppArgs
+      | throwError s!"Expected bind to have 4 arguments, found {<- e.getAppArgs.mapM (liftM ∘ ppExpr)}"
+    Utils.lambdaOne cont fun x body => do
+      let name ← x.fvarId!.getUserName
+      let contVal: m α := traverseProgram onResult onBind onBif body
+      onBind value name contVal
+  else if let .some bfInfo ← Bifurcation.Info.ofExpr e then
+    let contsTaggedVals <- 
+      bfInfo.branches.mapM fun br => do
+        Utils.lambdaTelescopeN br.toExpr br.numArgs fun xs body => do
+          let names ← xs.mapM (·.fvarId!.getUserName)
+          let other := traverseProgram onResult onBind onBif body
+          return (names, other)
+    onBif bfInfo contsTaggedVals
+  else
+    onResult e
+
+namespace ProgressStar
+
+structure Config where
+  preconditionTac: Option Syntax.Tactic := none
+
+structure Info where
+  script: Array Syntax.Tactic := #[]
+  unsolvedPreconditions: List MVarId := []
+
+instance: Append Info where
+  append inf1 inf2 := {
+    script := inf1.script ++ inf2.script,
+    unsolvedPreconditions := inf1.unsolvedPreconditions ++ inf2.unsolvedPreconditions,
+  }
+
+partial
+def evalProgressStar(cfg: Config): TacticM Info := withMainContext do
+  trace[ProgressStar] s!"Normalizing bind application in goal {←(getMainTarget >>= (liftM ∘ ppExpr))}"
+  Utils.simpAt (simpOnly := true) (thms := [``Aeneas.Std.bind_assoc_eq]) 
+    (loc := .targets #[] (type := true) )
+    (config := {}) (simprocs := []) (simpThms := [])
+    (declsToUnfold := []) (hypsToUse := [])  
+    <|> pure ()
+  let goalTy <- getMainTarget
+  trace[ProgressStar] s!"After bind normalization: {←ppExpr goalTy}"
+  let res ← Progress.programTelescope goalTy fun _xs _zs program _res _post => do
+    trace[ProgressStar] s!"Traversing {← ppExpr program}"
+    let resultName := .str .anonymous "res"
+    traverseProgram (onResult resultName) onBind onBif program
+  setGoals (res.unsolvedPreconditions ++ (←getGoals))
+  return res
+
+where
+  onResult resultName expr := do
```

Comment:
I don't fully understand this case: could you add a more detailed comment? (with maybe an example)